nixos/lib/make-ext4-fs.nix

```diff
@@ -82,10 +89,12 @@ pkgs.stdenv.mkDerivation {
         size=$(( blocks - ''${free##*:} + fudge ))
 
         echo "Resizing from $blocks blocks to $size blocks. (~ $((size*blocksize/1024/1024))MiB)"
-        EXT2FS_NO_MTAB_OK=yes resize2fs $out -f $size
+        EXT2FS_NO_MTAB_OK=yes resize2fs ./$img -f $size
       )
 
       # And a final fsck, because of the previous truncating.
-      fsck.ext4 -n -f $out
+      fsck.ext4 -n -f ./$img
+
+      bzip2 -c ./$img > $out
```

Comment:
Could we add a `echo "Compressing image..."` so the end-user knows it's not stuck at fsck? Especially considering the length of time it's actually compressing.